### ISSUE TITLE
Access arbitrary row in training data in mx.metric.custom

### ISSUE THREAD

**Issue by caprice-j** [dmlc/mxnet#1151]
Can I access arbitrary row of training data in mx.metric.custom( .., fucntion(){ < here > }  ) ?

I'd like to experiment hinge-like loss function, and I'd like to use values in the training data as thresholds.

However, when I tried the following code:

  demo.metric.myrange <- mx.metric.custom("range", function(label, pred) { 
    if( i_train_row > nrow(train_x) ){
      i_train_row <<- 1 # iteration restart
      i_train_itr <<- i_train_itr + 1
    }

   ....  processing ....

        mylog <<- paste0(mylog, paste0(i_train_row," label::", label[i], " pred::", round(pred[i],2),
                                       " lower::",max( train_x[i_train_row, c("hdev1", "hdev2", "hdev3")] ), " res::",res[i]," \n"))

    i_train_row <<- i_train_row + 1

 return(res)
)

I cannot access the correct train_x[i_train_row]. I guess that it caused from "sgd" options, but have no idea to deal with it. Any ideas?

**Comment by caprice-j** [comment]
My current (dirty) workaround is:

   train_y1  <<-    target_data_frame[ ,   "y" ]
   sysnoise <<- seq(.000001,.000001 * length(train_y1), by = .000001) 
   train_y  <<-   train_y1 + sysnoise

And then 

demo.metric.myrange <- mx.metric.custom("range", function(label, pred) { 

  i_train_row <- round( (label * 1000000 - as.integer(label) * 1000000), 0  ) 

  ...
    
}


It worked but seems too ad-hoc and might not work for far longer rows.
And the initial predict values look slightly biased towards the noised values...
